app/jobs/replace_upd_to_obj_upd_prohibited_job.rb

```diff
@@ -0,0 +1,51 @@
+class ReplaceUpdToObjUpdProhibitedJob < ApplicationJob
+  def perform(action:, rollback: false)
+    logger.info 'Ran ReplaceUpdToObjUpdProhibitedJob!'
+
+    start_adding_new_status_for_locked_domains(action: action, rollback: rollback)
+  end
+
+  private
+
+  def start_adding_new_status_for_locked_domains(action:, rollback:)
+    count = 0
+    Domain.all.find_in_batches do |domain_batches|
+      count += domain_batches.count
+      logger.info "Proccesing #{count} domains of #{Domain.count}"
+
+      domain_batches.each do |domain|
+        if domain.locked_by_registrant? && rollback
```

Comment:
Consider moving this block to private method which gets `domain` as an argument - thus we'll lower the overall complexity